adam-core/src/test/scala/org/bdgenomics/adam/converters/FragmentConverterSuite.scala

```diff
@@ -0,0 +1,168 @@
+    val read = reads(0)
+    assert(read.getSequence === "ACACACACTGTGTGAATTCCGGCCTTAA")
+    assert(read.getContig.getContigName === "ctg")
+    assert(read.getStart === 0L)
+    assert(read.getEnd === 28L)
+  }
+
+  sparkTest("convert an rdd of varied fragments from multiple contigs") {
+    val rdd = sc.parallelize(Seq(NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg1").build())
+      .setFragmentSequence("ACACACAC")
+      .setFragmentStartPosition(0L)
+      .build(), NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg1").build())
+      .setFragmentSequence("TGTGTG")
+      .setFragmentStartPosition(8L)
+      .build(), NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg1").build())
+      .setFragmentSequence("AATTCCGGCCTTAA")
+      .setFragmentStartPosition(14L)
+      .build(), NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg2").build())
+      .setFragmentSequence("ACACACAC")
+      .setFragmentStartPosition(0L)
+      .build(), NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg2").build())
+      .setFragmentSequence("AATTCCGGCCTTAA")
+      .setFragmentStartPosition(14L)
+      .build(), NucleotideContigFragment.newBuilder()
+      .setContig(Contig.newBuilder().setContigName("ctg3").build())
+      .setFragmentSequence("AATTCCGGCCTTAA")
+      .setFragmentStartPosition(14L)
+      .build()))
+
+    val reads = FragmentConverter.convertRdd(rdd)
+      .collect()
+
+    assert(reads.length === 4)
+
+    val ctg1Reads = reads.filter(_.getContig.getContigName == "ctg1")
+    assert(ctg1Reads.length === 1)
+
+    val ctg1Read = ctg1Reads.head
+    assert(ctg1Read.getSequence === "ACACACACTGTGTGAATTCCGGCCTTAA")
+    assert(ctg1Read.getContig.getContigName === "ctg1")
+    assert(ctg1Read.getStart === 0L)
+    assert(ctg1Read.getEnd === 28L)
+
+    val ctg2Reads = reads.filter(_.getContig.getContigName == "ctg2")
+    assert(ctg2Reads.length === 2)
+
+    val firstCtg2Read = ctg2Reads.filter(_.getStart == 0L).head
+    val secondCtg2Read = ctg2Reads.filter(_.getStart != 0L).head
+
+    assert(firstCtg2Read.getSequence === "ACACACAC")
+    assert(firstCtg2Read.getContig.getContigName === "ctg2")
+    assert(firstCtg2Read.getStart === 0L)
+    assert(firstCtg2Read.getEnd === 8L)
+    assert(secondCtg2Read.getSequence === "AATTCCGGCCTTAA")
+    assert(secondCtg2Read.getContig.getContigName === "ctg2")
+    assert(secondCtg2Read.getStart === 14L)
+    assert(secondCtg2Read.getEnd === 28L)
+
+    val ctg3Reads = reads.filter(_.getContig.getContigName == "ctg3")
+    assert(ctg3Reads.length === 1)
+
+    val ctg3Read = ctg3Reads.head
+    assert(ctg3Read.getSequence === "AATTCCGGCCTTAA")
+    assert(ctg3Read.getContig.getContigName === "ctg3")
+    assert(ctg3Read.getStart === 14L)
+    assert(ctg3Read.getEnd === 28L)
+  }
+}
```

Comment:
@massie for right now, we assume that we don't have overlapping contigs. That is mostly because we are importing from FASTA and we assume that contigs are unique.